### DISCUSSION TITLE
Docker container action with parametric Dockerfile filename

### DISCUSSION THREAD

**Discussion by uraimo** [community/community#26212] [Actions]
<p>I was playing around with the creation of a docker container action and I’ve noticed that I can’t select a specific Dockerfile (I have a few of them, each one in a different directory, more on this later) using some input variable provided by the user of the Action as shown below:</p>
<pre><code class="lang-auto">
# action.yml

name: 'para\_docker'

description: 'test'

inputs:

&amp;nbsp; type: # id of input

&amp;nbsp; description: 'type of the docker container'

&amp;nbsp; required: true

&amp;nbsp; default: 'default'

runs:

&amp;nbsp; using: 'docker'

&amp;nbsp; image: '${{ inputs.type }}/Dockerfile' \&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;

&amp;nbsp; args:

&amp;nbsp; &amp;nbsp; - ${{ inputs.type }}

</code></pre>
<p>I get an error here: “image:’${{ inputs.type }}/Dockerfile’”, because at that point (when the docker image is created before it’s launched) “inputs” does not exist.</p>
<p>I’m using the [path]/Dockerfile format because it seems to be the only one available, no custom Dockerfile names.</p>
<p>From this it seems that input variables can only be passed to args: and not much else, or at least they can’t be used for that specific parameter of the docker action type.</p>
<p>Is this correct and will this always stay like this?</p>
<p>And, can the name of the dockerfile be made parametric and dependent on some input parameter chosen by the user of the action in some other way?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/uraimo">@uraimo</a>,</p>
<p>Thanks for being here and our apologies for the delay. Currently, it is not possible to specify a value in that <code>image</code> field in the way that you’ve configured.</p>
<p>The only supported formats are documented here:</p>
<p><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#image" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#image</a></p>
<p>I’ve taken your suggestion and passed it along to the appropriate teams. Thanks again for reaching out :grinning:</p>

**Comment by uraimo** [comment]
<p>Thanks Andrea, being able to use variables in more places would be great!</p>
<p>In this case, I’ve worked around it launching Docker manually with a minimal Javascript action that boots the container with the Dockerfile I want.</p>

**Comment by mosteo** [comment]
<p>Hi <a class="mention" href="/uraimo">@uraimo</a> , could you elaborate a bit on how you do this? I’m trying to write an action that somehow takes the Docker image in which it has to run, but I’m stuck. Thanks!</p>

**Comment by uraimo** [comment]
<p>Hey <a class="mention" href="/mosteo">@mosteo</a>, you can take a look at this: <a href="https://github.com/uraimo/run-on-arch-action/tree/master/src" rel="nofollow noopener">https://github.com/uraimo/run-on-arch-action/tree/master/src</a><br>
My workaround uses that js launcher to execute the bash script where most of the logic is contained(setup the host, launch docker with the params obtained from the launcher), you could use the same approach.</p>
<p>Another thing that you’ll probably need is retrieving some data from inside your docker, remember that , unexpectedly, you will still be able to call things like ’echo ::set-output name=Name::Value‘ like you would inside the standard action container!</p>

**Comment by mosteo** [comment]
<p>Very cool. This certainly gives me several ideas to try. Thanks a lot!</p>
